File: java/55-Jump-Game.java
class Solution {

  public boolean canJump(int[] nums) {
    int goal = nums.length - 1;
    for (int i = nums.length - 2; i >= 0; i--) {
      if (nums[i] + i >= goal) {
        goal = i;
      }
    }
    return goal == 0;
  }
}

File: kotlin/55-Jump-Game.kt
class Solution {
    fun canJump(nums: IntArray): Boolean {
        var goal = nums.size - 1
        
        for (i in nums.size - 2 downTo 0) {
            if (i + nums[i] >= goal) {
                goal = i
            }
        }
        return goal == 0
    }
}